app/models/ems_refresh.rb

```diff
@@ -120,7 +120,9 @@ def self.get_target_objects(target, single_id = nil)
     targets_by_type = target.each_with_object(Hash.new { |h, k| h[k] = [] }) do |(target_class, id), hash|
       # Take care of both String or Class type being passed in
       target_class = target_class.to_s.constantize unless target_class.kind_of?(Class)
-      if [VmOrTemplate, Host, ExtManagementSystem, ManagerRefresh::Target].none? { |k| target_class <= k }
+
+      if ManagerRefresh::Inventory.persister_class_for(target_class).blank? &&
```

Comment:
Yeah this looks good, I'd love to get rid of `if [VmOrTemplate, Host, ExtManagementSystem, ManagerRefresh::Target].none?` since I don't think it makes sense to have that check here if it is provider dependent